tests/test_snmp_notification_receiver.rb

```diff
@@ -0,0 +1,134 @@
+    id = '4.5.6.7'
+    id2 = '8.9.10.11'
+
+    refute_includes(Cisco::SnmpNotificationReceiver.receivers, id)
+    refute_includes(Cisco::SnmpNotificationReceiver.receivers, id2)
+
+    receiver = Cisco::SnmpNotificationReceiver.new(id,
+                                                   true,
+                                                   'informs',
+                                                   '3',
+                                                   'priv',
+                                                   'ab',
+                                                   '45',
+                                                   'red',
+                                                   'ethernet1/3')
+
+    receiver2 = Cisco::SnmpNotificationReceiver.new(id2,
+                                                    true,
+                                                    'traps',
+                                                    '3',
+                                                    'auth',
+                                                    'cd',
+                                                    '46',
+                                                    'red',
+                                                    'ethernet1/4')
+
+    assert_includes(Cisco::SnmpNotificationReceiver.receivers, id)
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id], receiver)
+
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].source_interface,
+                 'ethernet1/3')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].port, '45')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].type, 'informs')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].username, 'ab')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].version, '3')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].vrf, 'red')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id].security, 'priv')
+
+    assert_includes(Cisco::SnmpNotificationReceiver.receivers, id2)
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2], receiver2)
+
+    assert_equal(
+      Cisco::SnmpNotificationReceiver.receivers[id2].source_interface,
+      'ethernet1/4')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].port, '46')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].type, 'traps')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].username, 'cd')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].version, '3')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].vrf, 'red')
+    assert_equal(Cisco::SnmpNotificationReceiver.receivers[id2].security,
+                 'auth')
+
+    receiver.destroy
+    receiver2.destroy
+    refute_includes(Cisco::SnmpNotificationReceiver.receivers, id)
+    refute_includes(Cisco::SnmpNotificationReceiver.receivers, id2)
+  end
```

Comment:
Is there no way to change any of the properties without doing a destroy and recreate? If I want to change the username for a receiver from 'ab' to 'cd', I have to call `instance.destroy()` and `instance.new()` rather than something like `instance.username = 'cd'`?